pkg/controllers/sync/controller.go

```diff
@@ -971,9 +966,9 @@ func (s *SyncController) ensureFinalizer(ctx context.Context, fedResource Federa
 	ctx, keyedLogger := logging.InjectLoggerValues(ctx, "finalizer-name", FinalizerSyncController)
 
 	obj := fedResource.Object()
-	isUpdated, err := finalizersutil.AddFinalizers(obj, sets.NewString(FinalizerSyncController))
-	if err != nil || !isUpdated {
-		return err
+	isUpdated := finalizersutil.AddFinalizers(obj, sets.NewString(FinalizerSyncController))
+	if !isUpdated {
+		return fmt.Errorf("failed to add finalizer to federated object")
```

Comment:
This looks wrong. Pls check again.

Reply:
The original logic is that if `err` is nil and `isUpdated` is false, the err (nil) will be returned here. So let me update it to match the original logic, thanks!